project/BUILD

```diff
@@ -23,13 +23,33 @@ java_library(
     deps = ["//third_party/java/protobuf:protobuf"],
 )
 
-java_plugin(
-    name = "dagger_query_plugin",
-    srcs = ["src/com/google/daggerquery/plugin/QueryPlugin.java"],
+java_library(
+    name = "dagger_query_plugin_sources",
+    srcs = glob(["src/com/google/daggerquery/plugin/**/*.java"]),
     deps = [
         "//:dagger-spi",
         ":dagger_query_protobuf",
         "//third_party/java/auto:service",
+        "//third_party/java/guava:guava",
+    ]
+)
+
+java_test(
+    name = "DaggerQueryTest",
+    size = "small",
+    srcs = glob(["tests/com/google/daggerquery/**/*.java"]),
```

Comment:
Please create a different test target per test. This allows tests to be run standalone.

This way you don't need to specify the test_class attribute, because the srcs will only contain a single test and is unambiguous.

Reply:
Done ✅

However it shows an error `cannot determine test class` and for some reason class name cannot be inferred using the target's name and its source-root-relative path, so I added `test_class`.

Reply:
Thanks!

Interesting, I'm not sure why the `cannot determine test class` error pops up, as the [docs](https://docs.bazel.build/versions/master/be/java.html#java_test.test_class) suggest the `test_class` will be inferred from the target's `name` and its source-root-relative path when omitted. In any case this is a really minor nit, so let's not get bogged down. :smiley: